nix/package.nix

```diff
@@ -117,6 +118,11 @@ in
 
       mkdir -p "$out/nix-support"
 
+      sed -i -e "s@^Exec=.*ghostty@Exec=$out/bin/ghostty@" $out/share/applications/com.mitchellh.ghostty.desktop
+      sed -i -e "s@^TryExec=.*ghostty@TryExec=$out/bin/ghostty@" $out/share/applications/com.mitchellh.ghostty.desktop
+      sed -i -e "s@^Exec=.*ghostty@Exec=$out/bin/ghostty@" $out/share/dbus-1/services/com.mitchellh.ghostty.service
+      sed -i -e "s@^ExecStart=.*ghostty@ExecStart=$out/bin/ghostty@" $out/lib/systemd/user/com.mitchellh.ghostty.service
+
```

Comment:
IIRC it's recommended that service files don't refer to the binary via an absolute path, since after a rebuild it's possible that the cached service could point to a GC'd and nonexistent store path

Reply:
For "normal" service files on a non-Nix system yes, but `nix/package.nix` is modifying the services files in the _installed NixOS package_. Just looking at my system with `grep ExecStart -- /etc/systemd/system/*` it's actually the other way around. It's vastly more common for a Nix service file to have an absolute path to the executable in `/nix/store` rather than relying on the executable being in the PATH.

The service files that are copied to `zig-out/share` rely on the binary being in the path.

Also, if the binary referred to by a service is GC'd, the service file itself would be GC'd and Nix would tell systemd to reload all of the service definitions.